Directory.Build.targets

```diff
@@ -0,0 +1,41 @@
+ <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+ 
+  <!-- Verifies if targets are included into Visual Studio CoreCLR.sln in project which outputs 
+       *.dll or *.exe and sets ShouldCopyVSBuildArtifacts to true  -->
+  <Target Name="_ShouldCopyVSBuildArtifacts" BeforeTargets="CopyVSBuildArtifacts;PostBuildEvent">
+
+    <PropertyGroup>
+      <ShouldCopyVSBuildArtifacts>false</ShouldCopyVSBuildArtifacts>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="($(TargetFileName.EndsWith('.dll')) Or $(TargetFileName.EndsWith('.exe'))) And $(SolutionName) == 'CoreCLR'">
+      <ShouldCopyVSBuildArtifacts >true</ShouldCopyVSBuildArtifacts>
+    </PropertyGroup>
+  </Target>
+
+  <!-- Copies *.dll and *.exe files built in Visual Studio to output directory and test layout directory -->
+  <Target Name="CopyVSBuildArtifacts" 
```

Comment:
When building from VS assemblies and executables are stored in `bin\obj\Windows_NT.$(Platform).$(Configuration)\...` output directories. These targets ensure that `.dll` and `.exe` are copied to `bin\Product\Windows_NT.$(Platform).$(Configuration)\` directory and `bin\tests\Windows_NT.$(Platform).$(Configuration)\Tests\Core_Root\` directory.

As a result it is possible to build modified native assemblies directly from VS and immediately start tests by running chosen XUnitWrapper.dll or individual test without need to hand copy assemblies. Generatelayout switch will handle newly built assemblies correctly since they are copied to `Product` output directory as well.

Reply:
Is there any reason to only do this copying while in VS? Why not do it all the time?